### DISCUSSION TITLE
Recommended setting for running vLLM for CPU

### DISCUSSION THREAD

**Discussion by jerin-metrum-ai** [vllm-project/vllm#5672] [Q&A]
What are the recommended settings for running vLLM on a CPU to achieve high performance? For instance, if I have a dual-socket server with 96 cores per socket, how many cores (--cpuset-cpus) should be allocated to run multiple replicas of vLLM?

**Comment by akhilreddy0703** [comment]
48 cores per instance would do fine, It's performing with almost 10 t/s throughput for single user.
[Echoswift](https://github.com/Infobellit-Solutions-Pvt-Ltd/EchoSwift/tree/main) is a performance benchmark tool for self hosted LLMs, currently supports TGI,vLLM,Llamacpp and Ollama
It's very useful to perform comparative tests to find out the best container size based on the latency and throughput.
